berkeley-mobile/Map/MapViewController.swift

```diff
@@ -400,10 +403,20 @@ extension MapViewController: SearchBarDelegate {
 extension MapViewController: SearchResultsViewDelegate {
     
     // drop new pin and show detail view on search
+    //var previousMapMarker: MapMarker?
+    //var previousPlaceMark: MapPlacemark?
     func choosePlacemark(_ placemark: MapPlacemark) {
         // remove last search pin
-        removeAnnotations(type: MapMarker.self)
-        removeAnnotations(type: SearchAnnotation.self)
+        
+        if((self.previousPlaceMark) != nil) {
```

Comment:
we prefer to use something like
`if let previousPlaceMark = self.previousPlaceMark`
here instead of checking nil then force unwrapping